contrib/!frameworks/react/packages.el

```diff
@@ -0,0 +1,231 @@
+;;; packages.el --- react Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2015 Sylvain Benner
+;; Copyright (c) 2014-2015 Andrea Moretti & Contributors
+;;
+;; Author: Andrea Moretti <axyzxp@gmail.com>
+;; URL: https://github.com/axyz
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;; List of all packages to install and/or initialize. Built-in packages
+;; which require an initialization must be listed explicitly in the list.
+(setq react-packages
+      '(
+        web-mode
+        js2-mode
+        js2-refactor
+        js-doc
+        json-mode
+        json-snatcher
+        flycheck
+        tern
+        jsfmt
+        company
+        company-tern
+        web-beautify
+        ))
+
+;; List of packages to exclude.
+(setq react-excluded-packages '())
+
+;; For each package, define a function react/init-<package-name>
+;;
+(defun react/post-init-flycheck ()
+  (add-hook 'react-mode-hook
+            (lambda ()
+              (with-eval-after-load 'flycheck
+                ;; use eslint with web-mode for jsx files
+                (flycheck-add-mode 'javascript-eslint 'react-mode)
+
+                ;; disable jshint since we prefer eslint checking
+                (setq-default flycheck-disabled-checkers
+                              (append flycheck-disabled-checkers
+                                      '(javascript-jshint)))
+
+                ;; disable json-jsonlist checking for json files
+                (setq-default flycheck-disabled-checkers
+                              (append flycheck-disabled-checkers
+                                      '(json-jsonlist)))))))
+
+(defun react/post-init-web-mode ()
+  (define-derived-mode react-mode web-mode "react")
+  (add-to-list 'auto-mode-alist '("\\.jsx\\'" . react-mode))
+  (add-to-list 'auto-mode-alist '("\\.react.js\\'" . react-mode))
+  (add-to-list 'magic-mode-alist '("/** @jsx React.DOM */" . react-mode))
+
+  (add-hook 'react-mode-hook
+            (lambda ()
+              (defadvice web-mode-highlight-part (around tweak-jsx activate)
+                (let ((web-mode-enable-part-face nil))
+                  ad-do-it)))))
+
+(defun react/init-tern ()
+  (use-package tern
+    :defer t
+    :config
+    (progn
+      (evil-leader/set-key-for-mode 'react-mode "mrrV" 'tern-rename-variable)
```

Comment:
I've kept it the same way it is on javascript layer to simplify eventual future diffs, however if it may impact on performance I will probably adopt the single function way